src/pages/itemListPage/AllItemListSection.js

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from "react";
+
+import ItemListHead from "./components/ItemListHead.js";
+import Item from "./components/Item.js";
+import PagenationBar from "../../components/PagenationBar.js";
+
+import { useMediaQuery } from "../../hooks/useMediaQuery.js";
+import { useApi } from "../../hooks/useApi.js";
+import { getProducts } from "../../apis/apis.js";
+import "./AllItemListSection.css";
+
+const pageSizeTable = { PC: 10, TABLET: 6, MOBILE: 4 };
+
+export default function AllItemListSection() {
+  const media = useMediaQuery();
+  const [paramObj, setParamObj] = useState({
+    page: 1,
+    pageSize: pageSizeTable[media],
+    orderBy: "recent",
+    keyword: undefined,
+  });
+  const [isLoading, error, data] = useApi(getProducts, paramObj);
+
+  useEffect(() => {
+    setParamObj((prevObj) => ({ ...prevObj, pageSize: pageSizeTable[media] }));
+  }, [media]);
+
+  return (
+    <section className="allItemList">
+      <ItemListHead setParamObj={setParamObj} />
+      {!isLoading && !error && (
+        <>
+          <div className="allItemList__body">
+            {data.list.map((item) => (
+              <Item key={item.id} data={item} />
+            ))}
+          </div>
+          <PagenationBar
+            totalPage={Math.ceil(data.totalCount / pageSizeTable[media])}
+            useParamObj={[paramObj, setParamObj]}
```

Comment:
P2:
paramObj라는 state는 해당 API와 관련된 모든 파람을 가지고 있기 때문에
pagination 컴포넌트에서는 불필요한 정보까지 알게 됩니다.
가능하면 해당 컴포넌트가 필요한 값만 넘기시는 걸 추천드립니다.

```suggestion
            currentPage={paramObj.page}
            onChange={onChangeCurrentPage}
```